spec/javascripts/models/beta/story_spec.js

```diff
@@ -862,4 +934,62 @@ describe('Story model', function () {
       expect(Story.cloneStory(story).documents).toEqual([]);
     });
   });
+
+  describe("possibleStatesFor", () => {
+    const invalidEstimates = [null, ''];
+    const invalidStoryTypes = [storyTypes.FEATURE]
+    const validEstimates = [1,2,3]
+    const validStoryTypes = [storyTypes.BUG, storyTypes.CHORE, storyTypes.RELEASE]
+
+    invalidStoryTypes.forEach(invalidStoryType => {
+      describe(`when storyType is ${invalidStoryType}`, () => {
+        invalidEstimates.forEach(invalidEstimate => {
+          describe(`when estimate is "${invalidEstimate}"`, () => {
+            it('returns all states', () => {
+              const story = { estimate: invalidEstimate, storyType: invalidStoryType}
+
+              expect(Story.possibleStatesFor(story).length).toEqual(7);
+            })
+          })
+
+          validEstimates.forEach(validEstimate => {
+            describe(`when estimate is ${validEstimate}`, () => {
+              it('return just the unscheduled state', () => {
+                const story = { estimate: validEstimate, storyType: invalidStoryType }
+
+                expect(Story.possibleStatesFor(story).length).toEqual(1);
+                expect(Story.possibleStatesFor(story)[0]).toEqual(status.UNSCHEDULED);
+              })
+            })
+          })
+        })
+      })
+    })
+
+    validStoryTypes.forEach(validStoryType => {
+      describe(`when storyType is ${validStoryType}`, () => {
+        invalidEstimates.forEach(invalidEstimate => {
+          describe(`when estimate is "${invalidEstimate}"`, () => {
```

Comment:
You can apply this style https://github.com/Codeminer42/cm42-central/pull/576/files#diff-e7cad8ffcafac32387e195a4957fe897R257-R273 here.